### ISSUE TITLE
微调所需显存大小

### ISSUE THREAD

**Issue by lizhipengpeng** [FlagAlpha/Llama2-Chinese#74]
显存48G的显卡来进行微调，仍然报显存不够，想问下微调13B的模型到底需要多少显存呀？

**Comment by wrenchonline** [comment]
deepspeed --include localhost:0,1 --master_port 29000 finetune_clm_lora.py \
    --model_name_or_path ../../meta-llama/Llama-2-7b-chat-hf \
    --train_files ../../data/train_sft.csv \
    --validation_files  ../../data/dev_sft.csv \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --do_train \
    --do_eval \
    --use_fast_tokenizer true \
    --output_dir ${output_model} \
    --evaluation_strategy  steps \
    --max_eval_samples 800 \
    --learning_rate 1e-4 \
    --gradient_accumulation_steps 8 \
    --num_train_epochs 5 \
    --warmup_steps 400 \
    --load_in_bits 4 \
    --lora_r 8 \
    --lora_alpha 16 \
    --target_modules q_proj,k_proj,v_proj,o_proj,down_proj,gate_proj,up_proj \
    --logging_dir ${output_model}/logs \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --preprocessing_num_workers 10 \
    --save_steps 200 \
    --eval_steps 200 \
    --save_total_limit 2000 \
    --seed 42 \
    --disable_tqdm false \
    --ddp_find_unused_parameters false \
    --block_size 1024 \
    --report_to tensorboard \
    --overwrite_output_dir \
    --ignore_data_skip true \
    --gradient_checkpointing \
    --ddp_timeout 18000000 \
    | tee -a ${output_model}/train.log
    
    
    试试这个参数。

**Comment by lizhipengpeng** [comment]
> 

这个参数也不太行，微调13B模型用48G显存还是会爆掉。

**Comment by wrenchonline** [comment]
我只测过7B的，这个参数刚好可以，没测过13B的一般13B全参数就是 13*4 = 52GB显存 ，在加上优化器大概 60-70G显存一定可以，如果加上fb16 就是除以2 大概 40 - 50 应该是够的。

**Comment by lizhipengpeng** [comment]
好的，感谢你的信息！

**Comment by pickwu** [comment]
7b的我用24G的显存，batch_size=4微调
